spec/models/report_spec.rb

```diff
@@ -120,31 +120,60 @@
     end
   end
 
+  describe '#unresolved_siblings?' do
+    subject { Fabricate :report }
+
+    context 'when the target account has other unresolved reports' do
+      before { Fabricate :report, action_taken_at: nil, target_account: subject.target_account }
+
+      it { is_expected.to be_unresolved_siblings }
+    end
+
+    context 'when the target account has a resolved report' do
+      before { Fabricate :report, action_taken_at: 3.days.ago, target_account: subject.target_account }
+
+      it { is_expected.to_not be_unresolved_siblings }
+    end
+
+    context 'when the target account has no other reports' do
+      before { described_class.where(target_account: subject.target_account).destroy_all }
+
+      it { is_expected.to_not be_unresolved_siblings }
+    end
+  end
+
   describe 'validations' do
+    subject { Fabricate.build :report }
+
     let(:remote_account) { Fabricate(:account, domain: 'example.com', protocol: :activitypub, inbox_url: 'http://example.com/inbox') }
+    let(:local_account) { Fabricate(:account, domain: nil) }
 
-    it 'is invalid if comment is longer than character limit and reporter is local' do
-      report = Fabricate.build(:report, comment: comment_over_limit)
-      expect(report.valid?).to be false
-      expect(report).to model_have_error_on_field(:comment)
+    context 'with a local reporter' do
+      subject { Fabricate.build :report, account: local_account }
+
+      it { is_expected.to_not allow_value(comment_over_limit).for(:comment) }
```

Comment:
Future improvement: lose all those "too big generating" methods, and stub the limits in specs.